code/game/machinery/hologram.dm

```diff
@@ -41,313 +37,275 @@ var/const/HOLOPAD_MODE = RANGE_BASED
 				return
-			if(last_request + 200 < world.time) //don't spam other people with requests either, you jerk!
-				last_request = world.time
-				var/obj/item/card/id/I = user.GetIdCard()
-				if(!I)
-					to_chat(user, SPAN_NOTICE("You need authorization to use the holocall system. Please equip a valid ID card."))
-					return
-				var/forcedcall = 0
-				if(access_heads in I.access) //Special functions for command level people
-					switch(alert(user,"Command level authorization detected. Additional functions available", "Command level menu", "Forced call","Regular call"))
-						if("Forced call")
-							forcedcall = 1
-				var/list/holopadlist = list()
-				for(var/obj/machinery/hologram/holopad/H in SSmachinery.processing_machines)
-					if((H.z in current_map.map_levels) && H.operable())
-						holopadlist["[H.loc.loc.name]"] = H	//Define a list and fill it with the area of every holopad in the world
-				holopadlist = sortAssoc(holopadlist)
-				var/temppad = input(user, "Which holopad would you like to contact?", "holopad list") as null|anything in holopadlist
-				targetpad = holopadlist["[temppad]"]
-				if(targetpad == src)
-					to_chat(user, "<span class='info'>Using such sophisticated technology, just to talk to yourself seems a bit silly.</span>")
-					return
-				if(targetpad)
-					make_call(targetpad, user, forcedcall)
-
-/obj/machinery/hologram/holopad/proc/make_call(var/obj/machinery/hologram/holopad/targetpad, var/mob/living/carbon/user, forcedcall)
-	targetpad.last_request = world.time
-	targetpad.sourcepad = src //This marks the holopad you are making the call from
-	targetpad.caller_id = user //This marks you as the caller
-	targetpad.incoming_connection = 1
-	playsound(targetpad.loc, 'sound/machines/chime.ogg', 25, 5)
-	targetpad.icon_state = "holopad1"
-	if(forcedcall)
-		targetpad.audible_message("<b>[src]</b> announces, \"Incoming call with command authorization from [targetpad.sourcepad.loc.loc].\"")
-		to_chat(user, "<span class='notice'>Establishing forced connection to the holopad in [targetpad.loc.loc]</span>")
-		targetpad.forced = 1
+			var/forced_call = FALSE
+			if(access_heads in I.access) //Special functions for command level people
+				switch(alert(user,"Command level authorization detected. Additional functions available.", "Command Level Menu", "Forced Call", "Regular Call"))
+					if("Forced Call")
+						forced_call = TRUE
+			var/list/holopadlist = list()
+			for(var/obj/machinery/hologram/holopad/H in SSmachinery.processing_machines - src)
+				if(ARE_Z_CONNECTED(H.z, z) && H.operable())
+					holopadlist["[H.holopad_id]"] = H	//Define a list and fill it with the area of every holopad in the world
+			holopadlist = sortAssoc(holopadlist)
+			var/chosen_pad = input(user, "Which holopad would you like to contact?", "Holopad List") as null|anything in holopadlist
+			if(!chosen_pad)
+				return
```

Comment:
I'd consider returning `last_request` to some low value for this case:

```suggestion
			if(!chosen_pad)
				last_request = world.time - 15 SECONDS
				return
```

(though TBF if this was supposed to look right the logic would have to be inverted to something like `pause_until` or whatever, or at least store the magic cooldown time in a define)